File: project_euler_23.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jun 19 17:57:16 2020

@author: Carlos Henrique
"""

abundant = []
cannot = []

for i in range(2, 28124):
    divisors = []
    for j in range(1, i):
        if i%j == 0:
            divisors.append(j)
    if sum(divisors) > i:
        abundant.append(i)
        
for i in range(2, 28124):
    abundant2 = []
    for item in abundant:
        if item < i:
            abundant2.append(item)
    for j in abundant2:
        if (all(i != j+k) for k in abundant2):
            if i not in cannot:
                cannot.append(i)

print("The sum of the numbers which cannot be written as the sum of two abundant numbers is", sum(cannot))

File: project_euler_23.jl
# project euler 23
abundant = []
cannot = []
can = []

for i in 2:28123
    divisors = []
    for j in 1:i
        if i%j == 0 && i != j
            append!(divisors, j)
        end
        if sum(divisors) > i
            append!(abundant, i)
        end
    end
end

for i in 2:28123
    for j in abundant
        for k in abundant
            if i == j+k
                append!(can, i)
            end
        end
    end
end
for i in 2:28123
    if i !in can
        append!(cannot, i)
    end
end

println("The sum of the numbers which cannot be written as the sum of two abundant numbers is", sum(cannot))